qa/rolling-upgrade/src/test/java/org/opensearch/upgrades/SystemIndicesUpgradeIT.java

```diff
@@ -115,4 +101,29 @@ public void testSystemIndicesUpgrades() throws Exception {
             });
         }
     }
+
+    /**
+     * Completed tasks get persisted into the .tasks index, so this method waits
+     * until the task is completed in order to verify that it has been successfully
+     * written to the index and can be retrieved.
+     */
+    private static void createAndVerifyStoredTask() throws Exception {
+        // Use update by query to create an async task
+        final Request updateByQueryRequest = new Request("POST", "/test_index_old/_update_by_query");
```

Comment:
Just curious, `_reindex` was replaced with `_update_by_query`, is there some particular reason?

Reply:
Because it is done twice, once for the old cluster and again for the updated cluster. Reindex creates a new index so you can't make the exact same request twice (unless you closed or deleted the new index). Update by query is simpler because it takes no parameters and doesn't create any new resources :)

Reply:
(Previously this step was only being done on the old cluster, so there was no problem. My change does this test for both the old and updated clusters.)